sdk/src/common/global_log_handler.cc

```diff
@@ -33,7 +33,7 @@ void DefaultLogHandler::Handle(LogLevel level,
   }
   output_s << std::endl;
   // TBD - print attributes
-  std::cout << output_s.str();  // thread safe.
+  std::cerr << output_s.str();  // thread safe.
```

Comment:
+1

Reply:
No, I don't think we should do this.

The application may print something in `std::cout`, opentelemetry-cpp adding anything there will just corrupt the application output.

All opentelemetry-cpp internal logs should go to the same place, otherwise it becomes impossible to use.

Reply:
But, the users have the option to disable the INFO and DEBUG, so the standard output is not cluttered. Ideally, users should not disable WARNING and ERROR messages, nor should these messages be directed to std::cout.

Reply:
I believe using `CustomLogHandler` is the right approach if user wants the output stream based on the specific log level. By default, the DefaultLogHandler should route `ERROR` and `WARN` messages to `std::cerr`, while directing `INFO` and `DEBUG` messages to `std::cout`.

Reply:
Sorry for spamming with multiple messages :) 

Another option could be to make the `DefaultLogHandler` customizable if the user wants to choose the output stream for each level. The default being `cout` for `INFO` and `DEBUG`, and `cerr` for `ERROR` and `WARNING`. Something like:

```cpp

class DefaultLogHandler : public LogHandler {
private:
    std::map<LogLevel, std::ostream*> log_streams;

public:
    DefaultLogHandler (
        std::ostream& error_stream = std::cerr,
        std::ostream& warn_stream = std::cerr,
        std::ostream& info_stream = std::cout,
        std::ostream& debug_stream = std::cout)
    {
        log_streams[LogLevel::Error] = &error_stream;
        log_streams[LogLevel::Warning] = &warn_stream;
        log_streams[LogLevel::Info] = &info_stream;
        log_streams[LogLevel::Debug] = &debug_stream;
    }

    void Handle(LogLevel level,
                const char *file,
                int line,
                const char *msg,
                const opentelemetry::sdk::common::AttributeMap &attributes) noexcept override {
              std::ostream* stream = log_streams[level]; 
              *stream << "[" << LevelToString(level) << "] ";
               //...
   }
}
```

Reply:
After looking at this again:

* For applications that implement a service that runs as a deamon, typically stdout is used to print information messages, like "service ready on port XXX", so it can make sense to log info and debug messages to stdout instead of stderr.
* For applications that are not deamons, that process stdin data and return results in stdout, printing anything in stdout will disrupt the application.

opentelemetry-cpp, and the default log handler, does not know which kind of application is instrumented, so the choice here is heuristic. A real application will most likely implement its own handler, to log according to severity, and to log into a different place anyway (dedicated application log file), so which stream we print data to by default has no real consequences.

Implemented the requested changes: info and debug to std::cout.

Reply:
> A real application will most likely implement its own handler
I think this is optimistic. Especially for applications not using otel logging. IMO having this go to different places depending on level is confusing. I don't see any reason why this would ever need to log to stdout.

Why not just make it configurable with an env variable? Or just log everything to stderr? Ideally with a default level of OFF